### PR TITLE
Pass dataset tuple as input to Keras model

### PR DESCRIPTION
**Pull request by dillondaudert**
This change allows passing nested tuples ((x_1, x_2, ...), (y_1, y_2, ...)) from a Dataset to a Keras model.

### CODE REVIEW

**Review by kuberiitb — APPROVED** [review]
Changes have implemented the suggestions. Should be merged.

**Review by kuberiitb — APPROVED** [review]
Commit has solved the issue and should be merged.

**Review by kuberiitb — APPROVED** [review]

**Comment by lgeiger** [comment]
This is very handy when using Keras together with `tf.data.Datasets` since tuples are used rather than lists.

Looks like the CI failure is unrelated.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Review by lgeiger — APPROVED** [review]

**Comment by lgeiger** [comment]
Looks like this was fixed on master as part of f5a7bea2b78c8a8b4b76060978369c3436b60b55

**Comment by qlzh727** [comment]
Is this PR still alive? If the issue has been fixed by other commit, we probably can close it here.

**Comment by dillondaudert** [comment]
Looks like this was fixed, as @lgeiger said. This can be closed.
